hazelcast-client/src/test/java/com/hazelcast/client/lock/LegacyClientConditionTest.java

```diff
@@ -0,0 +1,189 @@
+package com.hazelcast.client.lock;
+
+import com.hazelcast.client.test.TestHazelcastFactory;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.ICondition;
+import com.hazelcast.core.ILock;
+import com.hazelcast.spi.exception.DistributedObjectDestroyedException;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class LegacyClientConditionTest extends HazelcastTestSupport {
```

Comment:
I just renamed the original `ClientConditionTest`, because I wanted to use the name for the test inheriting from `ConditionAbstractTest`. 

In theory `LegacyClientConditionTest` could be removed as it appears all(?) cases are already covered by `ConditionAbstractTest`. However it feels funny to remove tests - there is always a risk there is case not covered in the `ConditionAbstractTest` <- maybe some subtle difference.